immunopepper/traversal.py

```diff
@@ -365,14 +365,20 @@ def get_spanning_index(coord, k):
     # consider the second junction
     if coord.start_v3 is None:
         spanning_id_range2 = [np.nan]
+        spanning_id_range1_2 = [np.nan]
     else:
         m2 = int((L1 + L2) / 3)
         if (L1 + L2) % 3 == 0:
             spanning_id_range2 = range(max(m2 - k + 1, 0), m2)
         else:
             spanning_id_range2 = range(max(m2 - k + 1, 0), m2 + 1)
+        if (spanning_id_range1 and spanning_id_range2):
+            spanning_id_range1_2 = [shared for shared in range(spanning_id_range2[0], spanning_id_range1[-1] + 1)
```

Comment:
why not use `set()` for `spanning_id_range*`? You could use it for finding intersections and later you anyway you do membership tests, which are much faster on sets.

Reply:
Good point, this will be faster